mmdeploy/codebase/mmcls/models/backbones/vision_transformer.py

```diff
@@ -0,0 +1,155 @@
+    Returns:
+        out (Tensor): A feature map output from InvertedResidual. The tensor
+        shape (N, Cout, H, W).
+    """
+    B = x.shape[0]
+    x, patch_resolution = self.patch_embed(x)
+
+    # cls_tokens = self.cls_token.expand(B, -1, -1)
+    x = torch.cat((self.cls_token, x), dim=1)
+    x = x + resize_pos_embed(
+        self.pos_embed,
+        self.patch_resolution,
+        patch_resolution,
+        mode=self.interpolate_mode,
+        num_extra_tokens=self.num_extra_tokens)
+    x = self.drop_after_pos(x)
+
+    if not self.with_cls_token:
+        # Remove class token for transformer encoder input
+        x = x[:, 1:]
+
+    outs = []
+    for i, layer in enumerate(self.layers):
+        x = layer(x)
+
+        if i == len(self.layers) - 1 and self.final_norm:
+            x = self.norm1(x)
+
+        if i in self.out_indices:
+            B, _, C = x.shape
+            if self.with_cls_token:
+                patch_token = x[:, 1:].reshape(B, *patch_resolution, C)
+                patch_token = patch_token.permute(0, 3, 1, 2)
+                cls_token = x[:, 0]
+            else:
+                patch_token = x.reshape(B, *patch_resolution, C)
+                patch_token = patch_token.permute(0, 3, 1, 2)
+                cls_token = None
+            if self.output_cls_token:
+                out = [patch_token, cls_token]
+            else:
+                out = patch_token
+            outs.append(out)
+
+    return tuple(outs)
+
+
+class MultiHeadAttentionop(torch.autograd.Function):
+    """Create onnx::MultiHeadAttention op."""
+
+    @staticmethod
+    def forward(ctx, q: Tensor, k: Tensor, v: Tensor, q_weight: Tensor,
+                q_bias: Tensor, k_weight: Tensor, k_bias: Tensor,
+                v_weight: Tensor, v_bias: Tensor, o_weight: Tensor,
+                o_bias: Tensor, embed_dims: int, num_heads: int) -> Tensor:
+        return torch.rand_like(q)
+
+    @staticmethod
+    def symbolic(g, q: Tensor, k: Tensor, v: Tensor, q_weight: Tensor,
+                 q_bias: Tensor, k_weight: Tensor, k_bias: Tensor,
+                 v_weight: Tensor, v_bias: Tensor, o_weight: Tensor,
+                 o_bias: Tensor, embed_dims: int, num_heads: int):
+
+        q_weight.setDebugName('q_weight')
+        q_bias.setDebugName('q_bias')
+
+        k_weight.setDebugName('k_weight')
+        k_bias.setDebugName('k_bias')
+
+        v_weight.setDebugName('v_weight')
+        v_bias.setDebugName('v_bias')
+
+        o_weight.setDebugName('o_weight')
+        o_bias.setDebugName('o_bias')
+
+        return g.op(
+            'mmdeploy::MultiHeadAttention',
+            q,
+            k,
+            v,
+            q_weight,
+            q_bias,
+            k_weight,
+            k_bias,
+            v_weight,
+            v_bias,
+            o_weight,
+            o_bias,
+            embed_dim_i=embed_dims,
+            num_heads_i=num_heads)
+
+
+@FUNCTION_REWRITER.register_rewriter(
```

Comment:
`multiheadattention__forward__ncnn` can be moved to `codebase/mmcls/models/utils`

Reply:
`multiheadattention` and FFN is main part of transformer, I donot think we should treat it as `utils`.

Reply:
The origin function is inside [models/utils](https://github.com/open-mmlab/mmclassification/blob/master/mmcls/models/utils/attention.py), I want to make sure the directory structure is the same as the origin repo. So other contributors can find it if they want to do any modification.

Reply:
And maybe this module could be used in other backbone/module in the future?

Reply:
done.